frontend/src/components/Schema.vue

```diff
@@ -1,92 +1,104 @@
+import {useRoute} from "vue-router";
 
-export default {
-  name: "Schema",
-  components: {BaseLayout, Tabbing},
-  data: function () {
-    return {
-      tabs: [
-        {
-          name: "Entities",
-          component: markRaw(EntityList),
-          icon: "table_view",
-          tooltip: "Show entities"
-        },
-        {
-          name: "Edit / Show",
-          component: markRaw(SchemaEdit),
-          icon: "mode_edit",
-          tooltip: "Edit/Show schema structure"
-        },
-        {
-          name: "Add Entity",
-          component: markRaw(EntityForm),
-          icon: 'add_circle',
-          tooltip: 'Add new entity'
-        },
-        {
-          name: "Permissions",
-          component: markRaw(PermissionList),
-          icon: "security",
-          tooltip: "Manage permissions on the schema"
-        },
-        {
-          name: "History",
-          component: markRaw(Changes),
-          icon: 'history',
-          tooltip: 'Change history of schema'
-        }
-      ],
-    }
+const {getSchema, activeSchema} = useSchema();
+const route = useRoute();
+const loading = ref(true);
+const tabs = ref([
+  {
+    name: "Entities",
+    component: markRaw(EntityList),
+    icon: "table_view",
+    tooltip: "Show entities"
   },
-  inject: ['activeSchema'],
-  computed: {
-    bindArgs() {
-      return [
-        { schema: this.activeSchema, advancedControls: true },
-        { schema: this.activeSchema },
-        { schema: this.activeSchema },
-        { objectType: "Schema", objectId: this.activeSchema?.id },
-        { schema: this.activeSchema },
-      ]
-    },
-    title() {
-      try {
-        return this.activeSchema.name;
-      } catch (e) {
-        return "n/a";
-      }
-    }
+  {
+    name: "Edit / Show",
+    component: markRaw(SchemaEdit),
+    icon: "mode_edit",
+    tooltip: "Edit/Show schema structure"
+  },
+  {
+    name: "Add Entity",
+    component: markRaw(EntityForm),
+    icon: 'add_circle',
+    tooltip: 'Add new entity'
+  },
+  {
+    name: "Permissions",
+    component: markRaw(PermissionList),
+    icon: "security",
+    tooltip: "Manage permissions on the schema"
   },
-  updated() {
-    if (this.activeSchema?.name) {
-        document.title = this.activeSchema.name;
-      }
+  {
+    name: "History",
+    component: markRaw(Changes),
+    icon: 'history',
+    tooltip: 'Change history of schema'
   }
+]);
+
+const bindArgs = computed(() => {
+  return [
+    {schema: activeSchema.value, advancedControls: true},
+    {schema: activeSchema.value},
+    {schema: activeSchema.value},
+    {objectType: "Schema", objectId: activeSchema.value?.id},
+    {schema: activeSchema.value},
+  ]
+});
+const title = computed(() => {
+  return activeSchema.value?.name || "n/a";
+});
+
+onUpdated(() => {
+  if (activeSchema.value?.name) {
+    document.title = activeSchema.value.name;
+  }
+});
+
+async function onSchemaUpdate() {
+  await getSchema();
```

Comment:
Similar to the watcher below, you might want to consider setting `loading.value` to `true` and `false` before and after the call to `getSchema` respectively :thinking:

Reply:
Yes, thank you!